File: frontend/src/api/document.ts
/**
 * 文档相关 API
 * CRUD、导出、重新整理
 */
import { get, post, put, del } from './client'
import { API_DOCUMENT } from '@/utils/constants'
import type { PaginatedData } from './client'

// ==================== 类型定义 ====================

/** 文档整理结果 */
export interface Document {
  id: number
  title: string
  original_title: string
  doc_type: string
  source_path: string
  source_url: string
  char_count: number
  ai_file_reading: boolean
  method: string
  model: string
  category: string
  summary: string
  overall_evaluation: string
  title_suggestion: string
  reason: string
  search_text: string
  tags: TagResponse[]
  sections: DocumentSection[]
  created_at: string
  updated_at: string
}

/** 标签 */
export interface TagResponse {
  id: number
  name: string
}

/** 文档章节 */
export interface DocumentSection {
  id: number
  section_title: string
  location_hint: string
  summary: string
  search_text: string
  created_at: string
}

/** 文档列表查询参数 */
export interface DocumentListParams {
  page?: number
  page_size?: number
  category?: string
  tag?: string
  doc_type?: string
  keyword?: string
}

/** 文档更新参数 */
export interface DocumentUpdateParams {
  title?: string
  category?: string
  tags?: string[]
  summary?: string
}

/** 批量删除参数 */
export interface BatchDeleteParams {
  ids: number[]
}

/** 导出参数 */
export interface ExportParams {
  ids?: number[]
  format?: 'json'
}

/** 文件 URL 响应 */
export interface FileUrlResponse {
  document_id: number
  doc_type: string
  title: string
  source_path: string
  source_url: string
  file_url: string | null
  location: string | null
  page_number: number | null
  video_start: number | null
  video_end: number | null
  is_available: boolean
}

/** 本地打开文件响应 */
export interface OpenLocallyResponse {
  opened: boolean
  method: string  // "local_file" | "url"
  path?: string
  url?: string
  doc_type: string
  location?: string | null
  page_number?: number | null
  video_start?: number | null
  video_end?: number | null
  sumatra_command?: string[] | Record<string, string[] | null>
}

// ==================== API 方法 ====================

/**
 * 获取整理结果列表（分页+筛选）
 */
export function getDocumentList(params: DocumentListParams = {}): Promise<PaginatedData<Document>> {
  return get<PaginatedData<Document>>(API_DOCUMENT.LIST, params as Record<string, unknown>)
}

/**
 * 获取单条整理结果详情
 */
export function getDocumentDetail(id: number): Promise<Document> {
  return get<Document>(API_DOCUMENT.DETAIL(id))
}

/**
 * 获取文档文件的访问 URL 和定位信息
 */
export function getDocumentFileUrl(id: number, location?: string): Promise<FileUrlResponse> {
  const params: Record<string, unknown> = {}
  if (location) params.location = location
  return get<FileUrlResponse>(API_DOCUMENT.FILE_URL(id), params)
}

/**
 * 本地打开文件（用系统默认应用）
 */
export function openDocumentLocally(id: number, location?: string): Promise<OpenLocallyResponse> {
  const params: Record<string, unknown> = {}
  if (location) params.location = location
  return post<OpenLocallyResponse>(API_DOCUMENT.OPEN_LOCALLY(id), undefined, { params })
}

/**
 * 更新整理结果
 */
export function updateDocument(id: number, data: DocumentUpdateParams): Promise<Document> {
  return put<Document>(API_DOCUMENT.DETAIL(id), data)
}

/**
 * 删除整理结果
 */
export function deleteDocument(id: number): Promise<void> {
  return del<void>(API_DOCUMENT.DETAIL(id))
}

/**
 * 批量删除整理结果
 */
export function batchDeleteDocuments(ids: number[]): Promise<void> {
  return post<void>(API_DOCUMENT.BATCH_DELETE, { ids } as BatchDeleteParams)
}

/**
 * 重新整理单条文档
 */
export function reorganizeDocument(id: number, provider?: string): Promise<{ task_id: number }> {
  return post<{ task_id: number }>(API_DOCUMENT.REORGANIZE(id), provider ? { provider } : {})
}

/**
 * 导出整理结果为 JSON
 */
export function exportDocuments(params: ExportParams = {}): Promise<Blob> {
  return post<Blob>(API_DOCUMENT.EXPORT, params, { responseType: 'blob' } as never)
}

File: backend/app/schemas/document.py
# -*- coding: utf-8 -*-
"""
文档相关 Pydantic Schema
"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


# ---- Section ----

class SectionBase(BaseModel):
    """章节基础信息"""
    section_title: str = Field(default="", description="章节标题")
    location_hint: str = Field(default="", description="位置提示")
    summary: str = Field(default="", description="章节摘要")
    search_text: str = Field(default="", description="章节语义检索内容")


class SectionResponse(SectionBase):
    """章节响应"""
    id: int
    document_id: int
    created_at: datetime

    model_config = {"from_attributes": True}


# ---- Tag ----

class TagResponse(BaseModel):
    """标签响应"""
    id: int
    name: str

    model_config = {"from_attributes": True}


# ---- Document ----

class DocumentBase(BaseModel):
    """文档基础信息"""
    title: str = Field(default="", description="整理后标题")
    original_title: str = Field(default="", description="原始标题")
    doc_type: str = Field(default="unknown", description="文档类型")
    source_path: str = Field(default="", description="原始文件路径")
    source_url: str = Field(default="", description="来源URL")
    char_count: int = Field(default=0, description="原始文本字符数")
    ai_file_reading: bool = Field(default=False, description="是否使用AI文件直读")
    method: str = Field(default="", description="整理方法")
    model: str = Field(default="", description="使用的LLM模型名")
    category: str = Field(default="其他", description="分类")
    summary: str = Field(default="", description="整体摘要")
    overall_evaluation: str = Field(default="", description="整体评价")
    title_suggestion: str = Field(default="", description="AI建议标题")
    reason: str = Field(default="", description="分类理由")
    search_text: str = Field(default="", description="全文语义检索内容")


class DocumentCreate(BaseModel):
    """创建文档请求"""
    title: str
    original_title: str = ""
    doc_type: str = "unknown"
    source_path: str = ""
    source_url: str = ""
    char_count: int = 0
    ai_file_reading: bool = False
    method: str = ""
    model: str = ""
    category: str = "其他"
    summary: str = ""
    overall_evaluation: str = ""
    title_suggestion: str = ""
    reason: str = ""
    search_text: str = ""
    tags: List[str] = Field(default_factory=list, description="标签列表")
    sections: List[SectionBase] = Field(default_factory=list, description="章节列表")


class DocumentUpdate(BaseModel):
    """更新文档请求"""
    title: Optional[str] = None
    category: Optional[str] = None
    summary: Optional[str] = None
    overall_evaluation: Optional[str] = None
    title_suggestion: Optional[str] = None
    reason: Optional[str] = None
    search_text: Optional[str] = None
    tags: Optional[List[str]] = None


class DocumentResponse(DocumentBase):
    """文档响应"""
    id: int
    created_at: datetime
    updated_at: datetime
    sections: List[SectionResponse] = Field(default_factory=list)
    tags: List[TagResponse] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class DocumentListItem(BaseModel):
    """文档列表项（不含大文本字段，用于列表展示）"""
    id: int
    title: str
    original_title: str
    doc_type: str
    source_path: str
    source_url: str
    char_count: int
    category: str
    summary: str = Field(default="", description="摘要截断显示")
    method: str
    model: str
    tags: List[TagResponse] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentFilter(BaseModel):
    """文档筛选参数"""
    keyword: Optional[str] = None
    category: Optional[str] = None
    doc_type: Optional[str] = None
    tag: Optional[str] = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)


class DocumentBatchDelete(BaseModel):
    """批量删除文档"""
    ids: List[int] = Field(..., min_length=1, description="待删除的文档ID列表")


class DocumentExportRequest(BaseModel):
    """导出请求"""
    ids: Optional[List[int]] = None
    category: Optional[str] = None


class ReorganizeRequest(BaseModel):
    """重新整理文档请求"""
    provider: str = Field(default="qwen", description="LLM提供者: qwen/deepseek")
